From 87dabe116a730cd1caf01d6ec139a91ce083bea0 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Wed, 25 Feb 2026 20:50:38 +0100
Subject: [PATCH 2/3] gluon-core: disable AQL on MT7915 radios

This is a workaround to stalling transmissions on MT7915 / MT7916 radios
due to stalling frames for station in PSM, breaking AQL assumptions.

Disable AQL for these radios. This reduces airtime fairness behavior but
prevents TX from stalling altogether.

Signed-off-by: David Bauer <mail@david-bauer.net>
---
 .../etc/hotplug.d/ieee80211/02-mt7915-aql-disable.sh  | 11 +++++++++++
 1 file changed, 11 insertions(+)
 create mode 100644 package/gluon-core/files/etc/hotplug.d/ieee80211/02-mt7915-aql-disable.sh

diff --git a/package/gluon-core/files/etc/hotplug.d/ieee80211/02-mt7915-aql-disable.sh b/package/gluon-core/files/etc/hotplug.d/ieee80211/02-mt7915-aql-disable.sh
new file mode 100644
index 00000000..8c7138f0
--- /dev/null
+++ b/package/gluon-core/files/etc/hotplug.d/ieee80211/02-mt7915-aql-disable.sh
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+[ "$ACTION" = "add" ] || exit 0
+
+PHYNBR=${DEVPATH##*/phy}
+
+[ -n "$PHYNBR" ] || exit 0
+
+[ -e "/sys/kernel/debug/ieee80211/phy${PHYNBR}/mt76/twt_stats" ] || exit 0
+
+echo 0 > "/sys/kernel/debug/ieee80211/phy${PHYNBR}/aql_enable"
-- 
2.51.0

